types/semver/semver-tests.ts

```diff
@@ -2,8 +2,10 @@ import * as semver from "semver";
 
 let bool: boolean;
 let num: number;
-let str = '';
-let strn: string | null = '';
+// $ExpectType string
+let str = String(Math.random());
+// $ExpectType string | null
+let strn: string | null = Math.random() < 0.5 ? null : str;
```

Comment:
Using&nbsp;`Math.random()` ensures that&nbsp;modern **TypeScript**&nbsp;versions won’t&nbsp;coerce these&nbsp;types&nbsp;to&nbsp;`''`, which&nbsp;they&nbsp;do.</p>